### ISSUE TITLE
validate fails on e2e stories

### ISSUE THREAD

**Issue by rgstephens** [RasaHQ/rasa#7695]
**Rasa version**: 2.2.3

**Rasa SDK version** (if used & relevant): 2.2.0

**Python version**: 3.8.5

**Operating system** (windows, osx, ...): osx

**Issue**:
`rasa data validate` fails on e2e story test with the following error:

**Error (including full traceback)**:
```
/rasa/shared/utils/io.py:93: UserWarning: The intent 'None' is used in your stories, but it is not listed in the domain file. You should add it to your domain file!
```

stories.yml:

```yml
- story: e2e test
  steps:
  - user: "what is your name"
  - bot: "my name is greg"
```

**Comment by TyDunn** [comment]
@JEM-Mosig Are you working on this is?

**Comment by TyDunn** [comment]
@JEM-Mosig Checking in on this because you assigned yourself to this. Are you taking this on?

**Comment by kedz** [comment]
I think this issue can be closed? The bug fix has been merged into `main` and `2.7.2`, and I have a [PR](https://github.com/RasaHQ/rasa/pull/9347) with the fix on `2.8.x` as well.
